infra/modules/network/vpc-endpoints.tf

```diff
@@ -37,6 +36,35 @@ data "aws_region" "current" {}
 # See https://repost.aws/knowledge-center/lambda-vpc-parameter-store
 # See https://docs.aws.amazon.com/vpc/latest/privatelink/create-interface-endpoint.html#create-interface-endpoint
 
+data "aws_subnet" "private" {
+  count = length(module.aws_vpc.private_subnets)
+  id    = module.aws_vpc.private_subnets[count.index]
+}
+
+# AWS services may only be available in certain regions and availability zones,
+# so we use this data source to get that information and only create
+# VPC endpoints in the regions / availability zones where the particular service
+# is available.
+data "aws_vpc_endpoint_service" "aws_service" {
+  for_each = local.interface_vpc_endpoints
+  service  = each.key
+}
+
+locals {
+  # Map from the name of an AWS service to a list of the private subnets that are in availability
+  # zones where the service is available. Only create this map for AWS services where we are going
+  # to create an Interface VPC endpoint, which require a list of subnet ids in which to create the
+  # elastic network interface for the endpoint.
+  aws_service_subnets = {
+    for service in local.interface_vpc_endpoints :
+    service => [
+      for subnet in data.aws_subnet.private[*] :
+      subnet.id
+      if contains(data.aws_vpc_endpoint_service.aws_service[service].availability_zones, subnet.availability_zone)
+    ]
+  }
+}
+
 resource "aws_security_group" "aws_services" {
```

Comment:
Something to keep in mind, AWS Security Group rule(s) quota is 60 inbound or outbound rules per SG. A request can be made to increase the quota, but if we begin to have too many entries it may be easier to create more than one SG so that consumers of the repo don't have to before using it. The quota for SGs is 2500 per Region.

https://docs.aws.amazon.com/vpc/latest/userguide/amazon-vpc-limits.html